Floofbot/Modules/RaidProtectionCommands.cs

```diff
@@ -0,0 +1,231 @@
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Floofbot.Modules
+{
+    [Summary("Raid protection configuration commands")]
+    [RequireUserPermission(GuildPermission.Administrator)]
+    [Group("raidconfig")]
+    public class RaidProtectionCommands : InteractiveBase
+    {
+        private FloofDataContext _floofDB;
+
+        public RaidProtectionCommands(FloofDataContext floofDB)
+        {
+            _floofDB = floofDB;
+        }
+        private Discord.Color GenerateColor()
+        {
+            return new Discord.Color((uint)new Random().Next(0x1000000));
+        }
+
+        private RaidProtectionConfig GetServerConfig(ulong server)
+        {
+            // checks if server exists in database and adds if not
+            var serverConfig = _floofDB.RaidProtectionConfigs.Find(server);
+            if (serverConfig == null)
+            {
+                _floofDB.Add(new RaidProtectionConfig
+                {
+                    ServerId = server,
+                    Enabled = false,
+                    ModChannelId = null,
+                    ModRoleId = null,
+                    MutedRoleId = null,
+                    BanOffenders = true,
+                    ExceptionRoleId = null
+                });
+                _floofDB.SaveChanges();
+                return _floofDB.RaidProtectionConfigs.Find(server);
+            }
+            else
+            {
+                return serverConfig;
+            }
+        }
+        private async Task<SocketRole> resolveRole(string input, SocketGuild guild, ISocketMessageChannel channel)
+        {
+            SocketRole role = null;
+            bool roleFound = false;
+            //resolve roleID or @mention
+            if (Regex.IsMatch(input, @"\d{10,}"))
+            {
+                string roleID = Regex.Match(input, @"\d{10,}").Value;
+                role = guild.GetRole(Convert.ToUInt64(roleID));
+                if (role != null)
+                    roleFound = true;
+                else
+                    roleFound = false;
+            }
+            //resolve role name
+            else
+            {
+                foreach (SocketRole r in Context.Guild.Roles)
+                {
+                    if (r.Name.ToLower() == input.ToLower())
+                    {
+                        if (roleFound == false) // ok we found 1 role thats GOOD
+                        {
+                            roleFound = true;
+                            role = r;
+                        }
+                        else // there is more than 1 role with the same name!
+                        {
+                            await channel.SendMessageAsync("More than one role exists with that name! Not sure what to do! Please resolve this. Aborting..");
+                            return null;
+                        }
+                    }
+                }
+            }
+            if (roleFound)
```

Comment:
Here you can always `return role` because can either be a legit role or `null`.

Reply:
done